1. Basic Web Development/index.html

```diff
@@ -23,63 +22,122 @@
                     </div>
-                    <div class="grid-element">
-                        <label for="read">Read Berserk by Kentaro Miura on your behalf</label>
-                        <input id="read" name="permissionAll" type="checkbox" value="read" />
+                    <div>
+                        <input id="permissionNssEdit" name="presetE" type="radio" value="Necessary" checked />
+                        <label for="permissionNssEdit">Necessary</label>
                     </div>
-                    <p class="agreeCountDisplay">You have agreed to</p>
-                    <div class="grid-element">
-                        <button type="submit" class="btnConfirm">Confirm</button>
-                        <a href="https://monkey-history.fandom.com/wiki/Water_Monke_/_Searching_For_The_Sea_Banana">Learn more</a>
+                    <h2>Permisison text</h2>
+                    <div class="text-input-element">
+                        <label for="permNameEdit">Display text</label>
+                        <input id="permNameEdit" name="permName" type="text" />
                     </div>
-                </form>             
+                    <p id="text-alert-edit" class="no-display">Text is required!</p>
+                    <button type="submit" class="btn-confirm">Confirm</button>
+                </form>
+            </div>
+            <div class="lock">
+                <input id="lock" name="lock" type="checkbox" checked />
+                <label for="lock">Lock Pop-up</label>
+            </div>
+        </div>
+        <div class="preferences hidden presentation">
+            <div class="bar"></div>
+            <div class="content">
+                <h1>Select your preferences:</h1>
+                <div class="section">
+                    <h2>Presets:</h2>
+                    <form>
+                        <div class="grid-element">
+                            <label for="all">All</label>
+                            <input id="all" name="preset" type="radio" value="All" />
+                        </div>
+                        <div class="grid-element">
+                            <label for="permissive">Permissive</label>
+                            <input id="permissive" name="preset" type="radio" value="Permissive" />
+                        </div>
+                        <div class="grid-element">
+                            <label for="necessary">Necessary</label>
+                            <input id="necessary" name="preset" type="radio" value="Necessary" checked />
+                        </div>
+                    </form>
+                </div>
+                <div class="section">
+                    <h2>Permissions:</h2>
+                    <form id="permissions-form">
+                        <div class="list">
+                            <div id="permissions-list"></div>
+                            <div id="spinner" class="spinner-grid no-display" >
```

Comment:
If you used a function to generate the loader, you wouldn't need to duplicate the code here.

Reply:
I created a function to generate it but I still need the first spinner in html for it to display while the js file is being fetched from the server.

Reply:
With async/defer tag, the script will be moved in the header and it begins to download a lot faster then at the end of the body. I see your point though.